LLVM 19 compatibility issues

We're currently updating LLVM to version 19 in the Arch Linux repositories (https://archlinux.org/todo/llvm-19/) and ispc is one of the last blocking packages. We have tried to compile the latest release and the latest master, both to no avail 🤔 

This is the error we're getting:
```
/build/ispc/src/ispc/src/opt/XeGatherCoalescePass.cpp:121:32: error: call to 'Create' is ambiguous
  121 |             {GEPVarOffsetData, llvm::GetElementPtrInst::Create(tType, tPtr, PartialIdxs, "partial_gep")});
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/llvm/IR/Instructions.h:938:29: note: candidate function
  938 |   static GetElementPtrInst *Create(Type *PointeeType, Value *Ptr,
      |                             ^
/usr/include/llvm/IR/Instructions.h:948:29: note: candidate function
  948 |   static GetElementPtrInst *Create(Type *PointeeType, Value *Ptr,
      |                             ^
```
It seems like your code just needs to select the right candidate from this overload set:
https://github.com/llvm/llvm-project/blob/e21dc4bd5474d04b8e62d7331362edcc5648d7e5/llvm/include/llvm/IR/Instructions.h#L938-L956

cc @tpkessler

--- a/examples/cpu/cmake/AddISPCExampleLegacy.cmake
+++ b/examples/cpu/cmake/AddISPCExampleLegacy.cmake
@@ -1,5 +1,5 @@
 #
-#  Copyright (c) 2018-2023, Intel Corporation
+#  Copyright (c) 2018-2025, Intel Corporation
 #
 #  SPDX-License-Identifier: BSD-3-Clause
 
@@ -12,7 +12,7 @@ function(add_ispc_example)
     set(multiValueArgs ISPC_IA_TARGETS ISPC_ARM_TARGETS ISPC_FLAGS TARGET_SOURCES LIBRARIES DATA_FILES)
     cmake_parse_arguments("example" "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
 
-    set(ISPC_KNOWN_TARGETS "sse2" "sse4" "avx1-" "avx2" "avx512knl" "avx512skx" "neon")
+    set(ISPC_KNOWN_TARGETS "sse2" "sse4" "avx1-" "avx2" "avx512skx" "neon")
     set(ISPC_HEADER_NAME "${CMAKE_CURRENT_BINARY_DIR}/${ISPC_SRC_NAME}_ispc.h")
     set(ISPC_OBJ_NAME "${CMAKE_CURRENT_BINARY_DIR}/${ISPC_SRC_NAME}.ispc${CMAKE_CXX_OUTPUT_EXTENSION}")
     set(ISPC_FLAGS ${example_ISPC_FLAGS})

--- a/examples/cpu/gmres/CMakeLists.txt
+++ b/examples/cpu/gmres/CMakeLists.txt
@@ -1,5 +1,5 @@
 #
-#  Copyright (c) 2018-2023, Intel Corporation
+#  Copyright (c) 2018-2025, Intel Corporation
 #
 #  SPDX-License-Identifier: BSD-3-Clause
 
@@ -16,7 +16,7 @@ set (TARGET_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/algorithm.cpp
                     ${CMAKE_CURRENT_SOURCE_DIR}/mmio.c
                     ${CMAKE_CURRENT_SOURCE_DIR}/mmio.h
                     ${CMAKE_CURRENT_SOURCE_DIR}/util.h)
-set (ISPC_IA_TARGETS "sse2-i32x4,sse4-i32x8,avx1-i32x16,avx2-i32x16,avx512knl-x16,avx512skx-x16" CACHE STRING "ISPC IA targets")
+set (ISPC_IA_TARGETS "sse2-i32x4,sse4-i32x8,avx1-i32x16,avx2-i32x16,avx512skx-x16" CACHE STRING "ISPC IA targets")
 set (ISPC_ARM_TARGETS "neon" CACHE STRING "ISPC ARM targets")
 add_ispc_example(NAME "gmres"
               ISPC_IA_TARGETS ${ISPC_IA_TARGETS}

--- a/ispcrt/cmake/ispc.cmake
+++ b/ispcrt/cmake/ispc.cmake
@@ -1,4 +1,4 @@
-## Copyright 2020-2023 Intel Corporation
+## Copyright 2020-2025 Intel Corporation
 ## SPDX-License-Identifier: BSD-3-Clause
 
 ###############################################################################
@@ -92,7 +92,6 @@ if (ISPC_X86_ENABLED)
   define_ispc_isa_options(SSE4 sse4-i32x4 sse4-i32x8 sse4-i16x8 sse4-i8x16)
   define_ispc_isa_options(AVX avx1-i32x8 avx1-i32x4 avx1-i32x16 avx1-i64x4)
   define_ispc_isa_options(AVX2 avx2-i32x8 avx2-i32x4 avx2-i32x16 avx2-i64x4)
-  define_ispc_isa_options(AVX512KNL avx512knl-x16)
   define_ispc_isa_options(AVX512SKX avx512skx-x16 avx512skx-x8)
   define_ispc_isa_options(AVX512SPR avx512spr-x16 avx512spr-x8)
 endif()
@@ -110,7 +109,6 @@ if (ISPC_X86_ENABLED)
   append_ispc_target_list(SSE4)
   append_ispc_target_list(AVX)
   append_ispc_target_list(AVX2)
-  append_ispc_target_list(AVX512KNL)
   append_ispc_target_list(AVX512SKX)
   if (NOT APPLE)
     append_ispc_target_list(AVX512SPR)

--- a/src/opt/XeGatherCoalescePass.cpp
+++ b/src/opt/XeGatherCoalescePass.cpp
@@ -1,5 +1,5 @@
 /*
-  Copyright (c) 2022-2024, Intel Corporation
+  Copyright (c) 2022-2025, Intel Corporation
 
   SPDX-License-Identifier: BSD-3-Clause
 */
@@ -117,8 +117,8 @@ MemoryCoalescing::BasePtrInfo MemoryCoalescing::analyseVarOffsetGEP(llvm::GetEle
             PartialIdxs.push_back(GEP->getOperand(i));
         llvm::Value *tPtr = GEP->getPointerOperand();
         llvm::Type *tType = GEP->getSourceElementType();
-        auto ret = ComplexGEPsInfoCache.insert(
-            {GEPVarOffsetData, llvm::GetElementPtrInst::Create(tType, tPtr, PartialIdxs, "partial_gep")});
+        auto ret =
+            ComplexGEPsInfoCache.insert({GEPVarOffsetData, llvm::GetElementPtrInst::Create(tType, tPtr, PartialIdxs)});
         DanglingGEP_it = ret.first;
     }
     llvm::Value *DanglingGEP = DanglingGEP_it->second;

--- a/src/opt/XeReplaceLLVMIntrinsics.cpp
+++ b/src/opt/XeReplaceLLVMIntrinsics.cpp
@@ -1,5 +1,5 @@
 /*
-  Copyright (c) 2022-2024, Intel Corporation
+  Copyright (c) 2022-2025, Intel Corporation
 
   SPDX-License-Identifier: BSD-3-Clause
 */
@@ -25,7 +25,7 @@ restart:
             if (func == nullptr || !func->isIntrinsic())
                 continue;
 
-            if (func->getName().equals("llvm.trap")) {
+            if (func->getName() == "llvm.trap") {
                 llvm::Type *argTypes[] = {LLVMTypes::Int1VectorType, LLVMTypes::Int16VectorType};
                 // Description of parameters for genx_raw_send_noresult can be found in target-genx.ll
                 auto Fn =
@@ -49,7 +49,7 @@ restart:
                     modifiedAny = true;
                     goto restart;
                 }
-            } else if (func->getName().equals("llvm.experimental.noalias.scope.decl")) {
+            } else if (func->getName() == "llvm.experimental.noalias.scope.decl") {
                 // These intrinsics are not supported by backend so remove them.
                 ci->eraseFromParent();
                 modifiedAny = true;

